library/VTKExtensions/Rendering/vtkF3DRenderer.cxx

```diff
@@ -1069,13 +1055,41 @@ void vtkF3DRenderer::Render()
 //----------------------------------------------------------------------------
 int vtkF3DRenderer::UpdateLights()
 {
+  // Recover the number of light that are on
   int lightCount = this->Superclass::UpdateLights();
+
+  // If no lights are turned on, add a light kit
   if (lightCount == 0 && !this->UseImageBasedLighting)
   {
     vtkNew<vtkLightKit> lightKit;
     lightKit->AddLightsToRenderer(this);
-    return this->GetLights()->GetNumberOfItems();
+    lightCount = this->GetLights()->GetNumberOfItems();
```

Comment:
extra nitpicky at that point but isn't that breaking the spec of the parent method now (returning the total light count instead of the "on" light count)? should be `lightCount += number_of_lights_in_the_lightkit` to be fully correct, even though I don't know if it's worth the trouble of calling `this->GetLights()->GetNumberOfItems()` before and after adding the lightkit to figure out the difference.

Reply:
Definitely correct although I dont see a case where it would matter. I will make the changes though.